frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx

```diff
@@ -1158,7 +1176,50 @@ const AttendanceRow = React.memo(function AttendanceRow({
               rowIndex={rowIndex}
               data-qa={`day-cell-${employeeId ?? ''}-${date.formatIso()}`}
             >
-              <DayCell>
+              <DayCell data-qa={`attendance-${date.formatIso()}-${rowIndex}`}>
+                <PlannedAttendanceTimes data-qa="planned-attendance-day">
+                  {plannedAttendancesForDate(date).length > 0 ? (
+                    plannedAttendancesForDate(date).map(
```

Comment:
Not a big deal, but all planned attendances are iterated twice here unnecessarily. You could just move this call to the `operationalDays.map` above.